src/common/types.hpp

```diff
@@ -231,6 +241,25 @@ class Ip6Address
      */
     bool IsLoopback(void) const { return (m32[0] == 0 && m32[1] == 0 && m32[2] == 0 && m32[3] == htobe32(1)); }
 
+    /**
+     * Returns the IPv6 address scope.
+     *
+     * @returns The IPv6 address scope.
+     */
+    uint8_t GetScope(void) const;
+
+    /**
+     * This function returns whether the address matches a given network prefix.
+     *
+     * This function only matches the prefix in unit of bytes, not bits.
+     *
+     * @param[in] aPrefix  The network prefix to check.
+     * @param[in] aLength  The length of the prefix to match (in bytes).
+     *
+     * @returns Whether this Ip6 address matches the prefix.
+     */
+    bool MatchPrefix(const uint8_t *aPrefix, uint8_t aLength) const { return memcmp(m8, aPrefix, aLength) == 0; }
```

Comment:
This may become confusing since prefix lengths are often specified in bits?

The similar method in the OT core `Ip6::Address` uses the bit length, so perhaps it would be better to harmonize this (to avoid incorrect use?)

[ot::Ip6::Address::MatchesPrefix()](https://github.com/openthread/openthread/blob/08a8767fcd8599178c40244c16935227c766ed1d/src/core/net/ip6_address.hpp#L775-L784)

Reply:
Updated! Used similiar implementation in OT and added a simple unit test.